lib/validation.js

```diff
@@ -64,6 +64,10 @@ function validateOpts (app, opts = {}) {
     if (typeof onError !== 'function') {
       throw new Error('onError must be a function')
     }
+  } else if (app.log && app.log.debug && typeof app.log.debug === 'function') {
+    onError = (prefix, fieldName, err) => {
+      app.log.debug({ err, msg: 'Mercurius cache error', prefix, fieldName })
+    }
```

Comment:
Can you please generate this closure in a top level function? So we can avoid closing over opts.

Reply:
Is a function factory the best option in this case? You can see it on the PR now:
```
function defaultDebug (app) {
  return function (prefix, fieldName, err) {
    app.log.debug({ err, msg: 'Mercurius cache error', prefix, fieldName })
  }
}
```